route-rs-runtime/src/link/async_link.rs

```diff
@@ -214,56 +269,105 @@ impl<Packet: Sized> Stream for AsyncEgressor<Packet> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::element::{AsyncIdentityElement, DropElement, IdentityElement, TransformElement};
+    use crate::element::{DropElement, IdentityElement, TransformElement};
     use crate::link::sync_link::SyncLinkBuilder;
-    use crate::link::{ElementLinkBuilder, LinkBuilder};
+    use crate::link::{ElementLinkBuilder, LinkBuilder, TokioRunnable};
     use crate::utils::test::packet_collectors::ExhaustiveCollector;
     use crate::utils::test::packet_generators::{immediate_stream, PacketIntervalGenerator};
     use core::time;
     use futures::future::lazy;
 
+    fn run_tokio(runnables: Vec<TokioRunnable>) {
+        tokio::run(lazy(|| {
+            for runnable in runnables {
+                tokio::spawn(runnable);
+            }
+            Ok(())
+        }));
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_input_streams() {
+        let identity_element: IdentityElement<i32> = IdentityElement::new();
+
+        AsyncLinkBuilder::new()
+            .element(identity_element)
+            .build_link();
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_element() {
+        let packets: Vec<i32> = vec![];
+        let packet_generator: PacketStream<i32> = immediate_stream(packets.clone());
+
+        AsyncLinkBuilder::<IdentityElement<i32>>::new()
+            .ingressor(packet_generator)
+            .build_link();
+    }
+
+    #[test]
+    fn builder_methods_work_in_any_order() {
+        let packets: Vec<i32> = vec![];
+
+        let packet_generator0 = immediate_stream(packets.clone());
+        let identity_element0 = IdentityElement::new();
+
+        AsyncLinkBuilder::new()
+            .ingressor(packet_generator0)
+            .element(identity_element0)
+            .build_link();
+
+        let packet_generator1 = immediate_stream(packets.clone());
+        let identity_element1 = IdentityElement::new();
+
+        AsyncLinkBuilder::new()
+            .element(identity_element1)
+            .ingressor(packet_generator1)
+            .build_link();
+    }
+
     #[test]
     fn async_link() {
```

Comment:
This test name should be more descriptive, or we can remove the example if it's not covering anything useful.

Reply:
doneish